File: lib/JFFUI/UITableView/UITableView+BlocksAdditions.m
#import "UITableView+BlocksAdditions.h"

@implementation UITableView (BlocksAdditions)

- (void)enumerateAllIndexPaths:(JFFVisitIndexPath)block
{
    NSParameterAssert(block);
    
    NSUInteger numberOfSections = [self numberOfSections];
    
    for (NSUInteger section = 0; section < numberOfSections; ++section) {
        
        @autoreleasepool {
            NSUInteger numberOfRows = [self numberOfRowsInSection:section];
            for (NSUInteger row = 0; row < numberOfRows; ++row) {
                
                @autoreleasepool {
                    NSIndexPath *indexPath = [NSIndexPath indexPathForRow:row inSection:section];
                    block(indexPath);
                }
            }
        }
    }
}

@end

File: iAsync.Swift/lib/JUI/JUI/UITableView/UITableView+BlocksAdditions.swift
import Foundation

import UIKit

public extension UITableView {
    
    func enumerateAllIndexPaths(@noescape block: (NSIndexPath) -> ())
    {
        let numberOfSections = self.numberOfSections()
        
        for section in 0..<numberOfSections {
            
            autoreleasepool {
                
                let numberOfRows = self.numberOfRowsInSection(section)
                
                for row in 0..<numberOfRows {
                    
                    autoreleasepool {
                        
                        let indexPath = NSIndexPath(forRow:row, inSection:section)
                        block(indexPath)
                    }
                }
            }
        }
    }
}
